### PR TITLE
Use create_contract host function in core tests

### PR DESCRIPTION
**Pull request by sisuresh**
# Description

This PR also fixes the wasm examples that broke due to changes in https://github.com/stellar/rs-stellar-contract-env.

# Checklist
- [ ] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [ ] Rebased on top of master (no merge commits)
- [ ] Ran `clang-format` v8.0.0 (via `make format` or the Visual Studio extension)
- [ ] Compiles
- [ ] Ran all tests
- [ ] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

**Review by jonjove — COMMENTED** [review]
A couple of very minor comments on this one. I would be comfortable merging it as is, so let me know what you think.

File: src/transactions/test/InvokeHostFunctionTests.cpp [hunk 1]
```diff
@@ -130,31 +131,74 @@ makeSymbol(std::string const& str)
...
 }
 
-template <typename T>
 static LedgerKey
-deployContract(AbstractLedgerTxn& ltx, T begin, T end)
+deployContract(Application& app, std::vector<uint8_t> const& contract,
+               uint256 const& salt, SecretKey const& key)
 {
+    // create signature
+    auto const& separator = "create_contract(nonce: u256, contract: Vec<u8>, "
```
[the remark this answers is not in the document]
Comment (sisuresh) [reply src/transactions/test/InvokeHostFunctionTests.cpp]: See response here - https://github.com/stellar/rs-stellar-contract-env/pull/144#discussion_r911294730.

File: src/transactions/test/InvokeHostFunctionTests.cpp [hunk 2]
```diff
@@ -35,43 +35,74 @@ using namespace stellar::txtest;
...
-    0x64, 0x00, 0x00, 0x0a, 0x5f, 0x5f, 0x64, 0x61, 0x74, 0x61, 0x5f, 0x65,
-    0x6e, 0x64, 0x03, 0x00, 0x0b, 0x5f, 0x5f, 0x68, 0x65, 0x61, 0x70, 0x5f,
-    0x62, 0x61, 0x73, 0x65, 0x03, 0x01, 0x0a, 0x47, 0x01, 0x45, 0x01, 0x02,
-    0x7f, 0x02, 0x40, 0x20, 0x00, 0x42, 0x0f, 0x83, 0x42, 0x03, 0x52, 0x20,
-    0x01, 0x42, 0x0f, 0x83, 0x42, 0x03, 0x52, 0x72, 0x45, 0x04, 0x40, 0x20,
-    0x01, 0x42, 0x04, 0x88, 0xa7, 0x22, 0x02, 0x41, 0x00, 0x48, 0x20, 0x02,
-    0x20, 0x00, 0x42, 0x04, 0x88, 0xa7, 0x22, 0x03, 0x6a, 0x22, 0x02, 0x20,
-    0x03, 0x48, 0x73, 0x45, 0x0d, 0x01, 0x0b, 0x00, 0x0b, 0x20, 0x02, 0xad,
-    0x42, 0x04, 0x86, 0x42, 0x03, 0x84, 0x0b};
+    0x00, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00, 0x00, 0x01, 0x0c, 0x02, 0x60,
```
[the remark this answers is not in the document]
Comment (sisuresh) [reply src/transactions/test/InvokeHostFunctionTests.cpp:43]: Nope I wasn't aware of `wasm-strip`. Updated the contracts and added instructions on `wasm-strip`.

File: src/transactions/test/InvokeHostFunctionTests.cpp [hunk 3]
```diff
@@ -35,43 +35,74 @@ using namespace stellar::txtest;
...
-    0x72, 0x79, 0x02, 0x00, 0x03, 0x70, 0x75, 0x74, 0x00, 0x02, 0x03, 0x64,
-    0x65, 0x6c, 0x00, 0x03, 0x0a, 0x5f, 0x5f, 0x64, 0x61, 0x74, 0x61, 0x5f,
-    0x65, 0x6e, 0x64, 0x03, 0x01, 0x0b, 0x5f, 0x5f, 0x68, 0x65, 0x61, 0x70,
-    0x5f, 0x62, 0x61, 0x73, 0x65, 0x03, 0x02, 0x0a, 0x35, 0x02, 0x1a, 0x01,
-    0x01, 0x7f, 0x23, 0x00, 0x41, 0x10, 0x6b, 0x22, 0x02, 0x24, 0x00, 0x20,
-    0x00, 0x20, 0x01, 0x10, 0x00, 0x20, 0x02, 0x41, 0x10, 0x6a, 0x24, 0x00,
-    0x0b, 0x18, 0x01, 0x01, 0x7f, 0x23, 0x00, 0x41, 0x10, 0x6b, 0x22, 0x01,
-    0x24, 0x00, 0x20, 0x00, 0x10, 0x01, 0x20, 0x01, 0x41, 0x10, 0x6a, 0x24,
-    0x00, 0x0b};
+    0x00, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00, 0x00, 0x01, 0x11, 0x03, 0x60,
```
[the remark this answers is not in the document]
Comment (sisuresh) [reply src/transactions/test/InvokeHostFunctionTests.cpp:73]: See comment above.

**Comment by jonjove** [comment]
r+ b4134a9

**Comment by jonjove** [comment]
r+ b4134a9

**Comment by jonjove** [comment]
@latobarita: retry

**Pull request merged**